keras2onnx/ke2onnx/conv.py

```diff
@@ -161,12 +162,22 @@ def convert_keras_conv_core(scope, operator, container, is_transpose, n_dims, in
                                                    op.dilation_rate,
                                                    list(range(
                                                        len(input_shape))) if channels_first else input_perm_axes)
+    elif op.padding == 'causal':
+        assert n_dims == 1
+        attrs['auto_pad'] = 'VALID'
+        left_pad = op.dilation_rate[0] * (op.kernel_size[0] - 1)
```

Comment:
Are you sure the padding is only related to the kernel size, irrelevant to the input?

Reply:
Yes, see the tensorflow backend [here](https://github.com/keras-team/keras/blob/master/keras/backend/tensorflow_backend.py#L3654). This pad is an extra one only used for causal mode, to do a preprocessing for causal. It is followed by Conv with another pad which is related to the kernel size